<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowBase</class>
 <widget class="QMainWindow" name="MainWindowBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BSDF Viewer</string>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>1</width>
     <height>1</height>
    </size>
   </property>
   <widget class="ViewerWidget" name="mainViewerWidget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>1</width>
      <height>1</height>
     </size>
    </property>
   </widget>
   <widget class="QDockWidget" name="controlDockWidget">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>10</y>
      <width>271</width>
      <height>231</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>250</width>
      <height>222</height>
     </size>
    </property>
    <property name="windowTitle">
     <string>Control</string>
    </property>
    <widget class="QWidget" name="dockWidgetContents_2">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QFormLayout" name="formLayout_2">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::ExpandingFieldsGrow</enum>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label_13">
          <property name="text">
           <string>Display mode:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="displayModeComboBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QFormLayout" name="formLayout_6">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::ExpandingFieldsGrow</enum>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Incoming polar angle:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QSlider" name="incomingPolarAngleSlider">
            <property name="pageStep">
             <number>1</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="incomingPolarAngleLineEdit">
            <property name="maximumSize">
             <size>
              <width>50</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>0</string>
            </property>
            <property name="maxLength">
             <number>6</number>
            </property>
            <property name="frame">
             <bool>true</bool>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="incomingAzimuthalAngleLabel">
          <property name="text">
           <string>Incoming azimuthal angle:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="wavelengthLabel">
          <property name="text">
           <string>Wavelength:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QSlider" name="wavelengthSlider">
            <property name="pageStep">
             <number>1</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="wavelengthLineEdit">
            <property name="maximumSize">
             <size>
              <width>50</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="frame">
             <bool>true</bool>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="1">
         <layout class="QHBoxLayout" name="incomingAzimuthalAngleLayout">
          <item>
           <widget class="QSlider" name="incomingAzimuthalAngleSlider">
            <property name="pageStep">
             <number>1</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="incomingAzimuthalAngleLineEdit">
            <property name="maximumSize">
             <size>
              <width>50</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>0</string>
            </property>
            <property name="maxLength">
             <number>6</number>
            </property>
            <property name="frame">
             <bool>true</bool>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="logPlotGroupBox">
        <property name="title">
         <string>Log plot</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
        <layout class="QFormLayout" name="formLayout_7">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::ExpandingFieldsGrow</enum>
         </property>
         <item row="0" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QSlider" name="logPlotBaseSlider">
             <property name="minimum">
              <number>2</number>
             </property>
             <property name="maximum">
              <number>99</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="tickPosition">
              <enum>QSlider::NoTicks</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="logPlotBaseLineEdit">
             <property name="maximumSize">
              <size>
               <width>30</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>10</string>
             </property>
             <property name="frame">
              <bool>false</bool>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>Base:</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>0</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QDockWidget" name="tableDockWidget">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>10</y>
      <width>271</width>
      <height>250</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>200</width>
      <height>250</height>
     </size>
    </property>
    <property name="baseSize">
     <size>
      <width>0</width>
      <height>500</height>
     </size>
    </property>
    <property name="windowTitle">
     <string>Table</string>
    </property>
    <widget class="QWidget" name="dockWidgetContents">
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="TableView" name="tableGraphicsView">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="mouseTracking">
         <bool>true</bool>
        </property>
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="backgroundBrush">
         <brush brushstyle="SolidPattern">
          <color alpha="255">
           <red>143</red>
           <green>143</green>
           <blue>143</blue>
          </color>
         </brush>
        </property>
        <property name="renderHints">
         <set>QPainter::TextAntialiasing</set>
        </property>
        <property name="dragMode">
         <enum>QGraphicsView::ScrollHandDrag</enum>
        </property>
        <property name="viewportUpdateMode">
         <enum>QGraphicsView::FullViewportUpdate</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QDockWidget" name="informationDockWidget">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>290</y>
      <width>271</width>
      <height>101</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>140</width>
      <height>94</height>
     </size>
    </property>
    <property name="windowTitle">
     <string>Information</string>
    </property>
    <widget class="QWidget" name="dockWidgetContents_4">
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <layout class="QFormLayout" name="formLayout_8">
        <item row="0" column="0">
         <widget class="QLabel" name="label_15">
          <property name="text">
           <string>Value:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="pickedValueLineEdit">
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="pickedReflectanceLabel">
          <property name="text">
           <string>Reflectance:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="pickedReflectanceLineEdit">
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>0</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QDockWidget" name="renderingDockWidget">
    <property name="geometry">
     <rect>
      <x>39</x>
      <y>280</y>
      <width>271</width>
      <height>341</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>250</width>
      <height>250</height>
     </size>
    </property>
    <property name="windowTitle">
     <string>Rendering</string>
    </property>
    <widget class="QWidget" name="dockWidgetContents_5">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_7">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="ViewerWidget" name="renderingViewerWidget" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formLayout_12">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
        </property>
        <property name="leftMargin">
         <number>9</number>
        </property>
        <property name="rightMargin">
         <number>9</number>
        </property>
        <property name="bottomMargin">
         <number>9</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label_22">
          <property name="text">
           <string>Light intensity:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="lightIntensitySpinBox">
          <property name="maximumSize">
           <size>
            <width>70</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>999.999000000000024</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>0.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_23">
          <property name="text">
           <string>Environment intensity:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="environmentIntensitySpinBox">
          <property name="maximumSize">
           <size>
            <width>70</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>999.999000000000024</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1000</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="fileMenu">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionOpenBrdf"/>
    <addaction name="separator"/>
    <addaction name="actionExport"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="viewMenu">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <widget class="QMenu" name="menuViewpoint">
     <property name="title">
      <string>Viewpoint</string>
     </property>
     <addaction name="actionViewFront"/>
     <addaction name="actionViewBack"/>
     <addaction name="actionViewRight"/>
     <addaction name="actionViewLeft"/>
     <addaction name="actionViewTop"/>
     <addaction name="actionViewBottom"/>
    </widget>
    <addaction name="separator"/>
    <addaction name="menuViewpoint"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="fileMenu"/>
   <addaction name="viewMenu"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="actionOpenBrdf">
   <property name="text">
    <string>&amp;Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionViewFront">
   <property name="text">
    <string>Front</string>
   </property>
   <property name="shortcut">
    <string>Home</string>
   </property>
  </action>
  <action name="actionViewBack">
   <property name="text">
    <string>Back</string>
   </property>
  </action>
  <action name="actionViewRight">
   <property name="text">
    <string>Right</string>
   </property>
  </action>
  <action name="actionViewLeft">
   <property name="text">
    <string>Left</string>
   </property>
  </action>
  <action name="actionViewTop">
   <property name="text">
    <string>Top</string>
   </property>
  </action>
  <action name="actionViewBottom">
   <property name="text">
    <string>Bottom</string>
   </property>
  </action>
  <action name="actionExport">
   <property name="text">
    <string>&amp;Export...</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TableView</class>
   <extends>QGraphicsView</extends>
   <header>TableView.h</header>
   <slots>
    <signal>redInfoUpdated(QString)</signal>
    <signal>greenInfoUpdated(QString)</signal>
    <signal>blueInfoUpdated(QString)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>ViewerWidget</class>
   <extends>QWidget</extends>
   <header>ViewerWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>displayModeComboBox</tabstop>
  <tabstop>incomingPolarAngleSlider</tabstop>
  <tabstop>incomingPolarAngleLineEdit</tabstop>
  <tabstop>wavelengthSlider</tabstop>
  <tabstop>wavelengthLineEdit</tabstop>
  <tabstop>incomingAzimuthalAngleSlider</tabstop>
  <tabstop>incomingAzimuthalAngleLineEdit</tabstop>
  <tabstop>logPlotGroupBox</tabstop>
  <tabstop>logPlotBaseSlider</tabstop>
  <tabstop>logPlotBaseLineEdit</tabstop>
  <tabstop>tableGraphicsView</tabstop>
  <tabstop>pickedValueLineEdit</tabstop>
  <tabstop>pickedReflectanceLineEdit</tabstop>
  <tabstop>lightIntensitySpinBox</tabstop>
  <tabstop>environmentIntensitySpinBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>incomingPolarAngleSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindowBase</receiver>
   <slot>updateIncomingPolarAngle(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>523</x>
     <y>113</y>
    </hint>
    <hint type="destinationlabel">
     <x>604</x>
     <y>106</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wavelengthSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindowBase</receiver>
   <slot>updateWavelength(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>523</x>
     <y>173</y>
    </hint>
    <hint type="destinationlabel">
     <x>601</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>logPlotBaseSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindowBase</receiver>
   <slot>updateBaseOfLogarithm(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>503</x>
     <y>226</y>
    </hint>
    <hint type="destinationlabel">
     <x>416</x>
     <y>417</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>logPlotGroupBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindowBase</receiver>
   <slot>useLogPlot(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>368</x>
     <y>207</y>
    </hint>
    <hint type="destinationlabel">
     <x>435</x>
     <y>454</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>displayModeComboBox</sender>
   <signal>activated(QString)</signal>
   <receiver>MainWindowBase</receiver>
   <slot>updateDisplayMode(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>487</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>604</x>
     <y>519</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lightIntensitySpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindowBase</receiver>
   <slot>updateLightIntensity(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>236</x>
     <y>606</y>
    </hint>
    <hint type="destinationlabel">
     <x>63</x>
     <y>601</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>environmentIntensitySpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindowBase</receiver>
   <slot>updateEnvironmentIntensity(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>236</x>
     <y>632</y>
    </hint>
    <hint type="destinationlabel">
     <x>182</x>
     <y>616</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>incomingAzimuthalAngleSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindowBase</receiver>
   <slot>updateIncomingAzimuthalAngle(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>523</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>632</x>
     <y>126</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateIncomingPolarAngle(int)</slot>
  <slot>createBrdfFromReflectanceModel()</slot>
  <slot>updateWavelength(int)</slot>
  <slot>updateBaseOfLogarithm(int)</slot>
  <slot>useLogPlot(bool)</slot>
  <slot>updateReflectanceModelCoordinateSystem(QString)</slot>
  <slot>showSamplePoint(bool)</slot>
  <slot>showSamplePointLabel(bool)</slot>
  <slot>updateDisplayMode(QString)</slot>
  <slot>updateLightIntensity(double)</slot>
  <slot>updateEnvironmentIntensity(double)</slot>
  <slot>updateIncomingAzimuthalAngle(int)</slot>
  <slot>changeCookTorranceButtonColor()</slot>
  <slot>changeLambertianButtonColor()</slot>
  <slot>changePhongButtonColor()</slot>
  <slot>changeWardAnisotropicButtonColor()</slot>
  <slot>changeWardIsotropicButtonColor()</slot>
 </slots>
</ui>
